### PR TITLE
Get partitions method

### PR DESCRIPTION
**Pull request by OleksiiZubko**
* Add the new QueryApi client for fetching Quad Tree Index
* Add getPartitions() by PartitionsRequest for Versioned Layer
* Add getPartitions() by QuadKeyPartitionsRequest  for Versioned Layer
* Add getPartitions() by PartitionsRequest for Volatile Layer
* Add getPartitions() by QuadKeyPartitionsRequest  for Volatile Layer
* Optimise imports

Resolves: OLPEDGE-930, OLPEDGE-937

### CODE REVIEW

File: @here/olp-sdk-dataservice-read/lib/client/PartitionsRequest.ts [hunk 1]
```diff
@@ -17,24 +17,45 @@
...
 export class PartitionsRequest {
-    private version: number | undefined;
+    private version?: number;
+    private quadKey?: QuadKey;
 
     public getVersion(): number | undefined {
         return this.version;
     }
 
+    public getQuadKey(): QuadKey | undefined {
```
Comment (andescu) [inline @here/olp-sdk-dataservice-read/lib/client/PartitionsRequest.ts]: As far as i remeber we said we will separate the QuadKey partitions request from the one with only partitions IDs:

- getPartitions(QuadKeyPartitionsRequest) with QuadKeyPartitionsRequest().withQuadKey(quadKey).withVersion(version).withDepth(depth)

- getPartitions(PartitionsRequest) with PartitionsRequest().withVersion(version).withPartitionIds(list)

File: @here/olp-sdk-dataservice-read/lib/client/PartitionsRequest.ts [hunk 2]
```diff
@@ -31,7 +31,8 @@ export class PartitionsRequest {
      * Setter for the provided version.
      * Method is optional. If the version does not specified the last layer version will be used
      * @param version Required. Specify the layer version.
-     * @returns this to have ability to chain methods
+     *
+     * @returns The updated [[PartitionsRequest]] instance
      */
     public withVersion(version: number): PartitionsRequest {
```
Comment (andescu) [inline @here/olp-sdk-dataservice-read/lib/client/PartitionsRequest.ts]: Allow also optional

File: @here/olp-sdk-dataservice-read/lib/client/QuadKeyPartitionsRequest.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    private quadKey?: QuadKey;
+    private depth?: QuadTreeIndexDepth;
+
+    /**
+     * The version of the catalog against which to run the query.
+     * Must be a valid catalog version.
+     * @param version A valid catalog version. The latest catalog version uses if not set.
+     * @returns The updated [[QuadKeyPartitionsRequest]] instance
+     */
+    public withVersion(version: number): QuadKeyPartitionsRequest {
```
Comment (andescu) [inline @here/olp-sdk-dataservice-read/lib/client/QuadKeyPartitionsRequest.ts]: Allow also optional

File: @here/olp-sdk-dataservice-read/lib/client/PartitionsRequest.ts [hunk 4]
```diff
@@ -30,10 +30,11 @@ export class PartitionsRequest {
     /**
      * Setter for the provided version.
      * Method is optional. If the version does not specified the last layer version will be used
-     * @param version Required. Specify the layer version.
-     * @returns this to have ability to chain methods
+     * @param version Specify the layer version. The latest catalog version uses if not set.
```
Comment (andescu) [inline @here/olp-sdk-dataservice-read/lib/client/PartitionsRequest.ts]:
```suggestion
     * @param version Specify the catalog version or set to undefined to use the latest catalog version.
```

File: @here/olp-sdk-dataservice-read/lib/client/QuadKeyPartitionsRequest.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+ */
+export class QuadKeyPartitionsRequest {
+    private version?: number;
+    private quadKey?: QuadKey;
+    private depth?: QuadTreeIndexDepth;
+
+    /**
+     * The version of the catalog against which to run the query.
+     * Must be a valid catalog version.
+     * @param version Optional valid catalog version. The latest catalog version uses if not set.
```
Comment (andescu) [inline @here/olp-sdk-dataservice-read/lib/client/QuadKeyPartitionsRequest.ts]:
```suggestion
     * @param version Specify the catalog version or set to undefined to use the latest catalog version.
```

File: @here/olp-sdk-dataservice-read/lib/client/PartitionsRequest.ts (hunk 4 above)
Comment (OleksiiZubko) [reply @here/olp-sdk-dataservice-read/lib/client/PartitionsRequest.ts]: updated, Thanks.

File: @here/olp-sdk-dataservice-read/lib/client/QuadKeyPartitionsRequest.ts (hunk 5 above)
Comment (OleksiiZubko) [reply @here/olp-sdk-dataservice-read/lib/client/QuadKeyPartitionsRequest.ts]: updated, thanks

**Pull request merged**